File: 2020/day01/puzzle1.py
file = open('puzzle1.in')
expenses = []
target_sum = 2020

for line in file:
  expense = int(line)
  expenses.append(expense)

def part1(expenses):
  for i in range(0, len(expenses) - 1):
    for j in range(i + 1, len(expenses)):
      if expenses[i] + expenses[j] == target_sum:
        print(expenses[i] * expenses[j])
        return

def part2(expenses):
  for i in range(0, len(expenses) - 1):
    for j in range(i + 1, len(expenses)):
      for k in range(j + 1, len(expenses)):
        if expenses[i] + expenses[j] + expenses[k] == target_sum:
          print(expenses[i] * expenses[j] * expenses[k])
          return

part1(expenses)
part2(expenses)

File: 2021/day01/puzzle1.js
const fs = require('fs');

const input = fs.readFileSync("puzzle1.in", "utf-8");
const depths = input.split("\n").map((line) => parseInt(line));

function part1(depths) {
  const increases = depths.reduce((count, depth, index, array) =>
    count + (array[index] < (array[index + 1] || 0) ? 1 : 0), 0);

  console.log(increases);
}

function part2(depths) {
  const windows = depths.map((depth, index, array) =>
    array[index] + (array[index + 1] || 0) + (array[index + 2] || 0));

  part1(windows);
}

part1(depths);
part2(depths);